src/UI/GameOptionSetupController.java

```diff
@@ -40,6 +50,32 @@ public GameOptionSetupController(GameBuilder aGameBuilder)
 
     gameOptions = new GameOption<?>[] {fowOption, startingFundsOption, incomeOption, weatherOption, unitSchemeOption, tagsOption, securityOption};
     optionSelector = new OptionSelector( gameOptions.length );
+
+    // Read in the last settings we used on this map, if available
+    initialPicksMap = new HashMap<Integer, String>();
+    ConfigUtils.readConfigLists(PlayerSetupController.buildSettingsFileName(gameBuilder),
+                                (String s)->Integer.valueOf(s),
+                                (Scanner linescan)->linescan.nextLine(),
+                                initialPicksMap);
+
+    if( initialPicksMap.containsKey(GAME_OPTIONS_CONFIG_KEY) )
+    {
+      String fullConfig = initialPicksMap.get(GAME_OPTIONS_CONFIG_KEY);
+      String[] s = fullConfig.split("\\s+"); // Split (string integers)
+      int si = 0; // Split Index
+      while (s[si].isEmpty()) ++si; // Skip any sneaky empty splits
+
+      for( int i = 0; i < gameOptions.length; ++i )
+        if( s.length > si )
+          gameOptions[i].setSelectedOption(Integer.valueOf(s[si++]));
+    }
+    else
+    {
+      if( gameBuilder.mapInfo.mapName.startsWith("HF") )
+        incomeOption.setSelectedOption(3 + 4); // 250 = 0, increment 250
```

Comment:
Why 3+4 instead of 7? Is 4 the normal setting and 7 is the normal HF setting?

Reply:
250*(1+3) is the normal 1k setting, so yes.

Reply:
There you go with the 1+3 instead of 4 thing again. makes no sense. If you want to break it out this way, say why.

In fact, if you want to use magic numbers at all, say why.